backend/controllers/permissions_functions.py

```
@@ -81,6 +84,50 @@ async def update_role(payload: RoleUpdateInput, request:Request):
+        users_with_this_role = await UserHandler.find_users_with_role_id(payload.id)
+        
+        for user_with_role in users_with_this_role:
+            # Simulate what permissions would be after the role update
+            current_role_ids = user_with_role.get('roles', [])
+            # We need to check what permissions they'd have with the updated role
+            updated_role_permissions = {}
+            for key in RoleHandler.permission_template:
+                updated_role_permissions[key] = getattr(payload, key, False)
+            
+            # Check if user would lose admin permissions
+            temp_perms = await RoleHandler.infer_permissions(current_role_ids)
```

Ruff (0.14.3)

Local variable `temp_perms` is assigned to but never used

Remove assignment to unused variable `temp_perms`

(F841)